app/graphql/queries/consultation/profile.rb

```diff
@@ -7,7 +7,10 @@ class Profile < Queries::BaseQuery
 	    type Types::Objects::Consultation::Base, null: true
 
 	    def resolve(id:)
-	    	::Consultation.find(id)
+	    	consultation = ::Consultation.find(id)
+	    	return consultation if consultation.public_consultation?
+	    	return consultation if context[:current_user].present? 
+	    	raise CivisApi::Exceptions::Unauthorized
```

Comment:
@BALAJI-BALA - I think we can simplify it to 

`raise CivisApi::Exceptions::Unauthorized if (!context[:current_user].present? && consultation.private_consultation?)
return consultation
`